api/src/utils/nova/searchQueryBuilder.ts

```diff
@@ -0,0 +1,155 @@
+
+    /**
+     * The query string in lower case.
+     */
+    private readonly queryLower: string;
+
+    /**
+     * Thed human readable part to use for bech32.
+     */
+    private readonly networkBechHrp: string;
+
+    /**
+     * Creates a new instance of SearchQueryBuilder.
+     * @param query The query string.
+     * @param networkBechHrp The network bechHrp.
+     */
+    constructor(query: string, networkBechHrp: string) {
+        this.query = query;
+        this.queryLower = query.toLowerCase();
+        this.networkBechHrp = networkBechHrp;
+    }
+
+    /**
+     * Builds the SearchQuery.
+     * @returns the SearchQuery object.
+     */
+    public build(): SearchQuery {
+        let address: IAddressDetails;
+        let blockId: string;
+        let transactionId: string;
+        let outputId: string;
+        let accountId: string;
+        let nftId: string;
+        let foundryId: string;
+        let anchorId: string;
+        let delegationId: string;
+        let tag: string;
+
+        const queryDetails = AddressHelper.buildAddress(this.networkBechHrp, this.queryLower);
+        // if the source query was valid bech32, we should directly look for an address
+        if (queryDetails.bech32) {
+            address = queryDetails;
+        } else {
+            // if the hex has 74 characters it might be a block id or a tx id
+            if (queryDetails?.hex && queryDetails.hex.length === 74) {
+                blockId = queryDetails.hex;
+                transactionId = queryDetails.hex;
+            } else if (queryDetails?.hex && queryDetails.hex.length === 66) {
+                // if the hex has 66 characters it might be a accoount id or a nft id
+                accountId = queryDetails.hex;
+                nftId = queryDetails.hex;
+                anchorId = queryDetails.hex;
+                delegationId = queryDetails.hex;
+            } else if (
+                // if the hex without prefix is 76 characters and first byte is 08,
+                // it might be a FoundryId (tokenId)
+                queryDetails.hex &&
+                Converter.isHex(queryDetails.hex, true) &&
+                queryDetails.hex.length === 78 &&
+                Number.parseInt(HexHelper.stripPrefix(queryDetails.hex).slice(0, 2), 16) === AddressType.Account
+            ) {
+                foundryId = queryDetails.hex;
+            } else if (
+                // if the hex is 70 characters it might be an outputId
+                queryDetails?.hex &&
+                queryDetails.hex.length === 78
+            ) {
+                outputId = queryDetails.hex;
+            }
+
+            // also perform a tag search
+            const maybeTag = Converter.isHex(this.query, true) ? HexHelper.addPrefix(this.query) : Converter.utf8ToHex(this.query, true);
+            if (maybeTag.length < 66) {
+                tag = maybeTag;
+            }
+        }
+
+        return {
+            queryLower: this.queryLower,
```

Comment:
if its an integer, we could redirect to `slot`, we can treat them in the same way we treat milestones now in stardust and chrysalis

Reply:
Sure :+1:  But we will also need a new task to implement the search improvements for slots, epochs, delegation...

Reply:
Created the task :point_right: https://github.com/iotaledger/explorer/issues/1117